gooddata-sdk/gooddata_sdk/catalog.py

```diff
@@ -17,99 +17,55 @@
 
 
 class CatalogEntry:
+    def __init__(self, entity: dict[str, Any]) -> None:
+        self._e = entity["attributes"]
+        self._entity = entity
+        self._obj_id = ObjId(self._entity["id"], type=self._entity["type"])
+
     @property
     def id(self) -> str:
-        raise NotImplementedError()
+        return self._entity["id"]
 
     @property
     def type(self) -> str:
-        raise NotImplementedError()
-
-    @property
-    def obj_id(self) -> ObjId:
-        raise NotImplementedError()
+        return self._entity["type"]
 
     @property
     def title(self) -> str:
-        raise NotImplementedError()
+        return self._e.get("title")
 
     @property
     def description(self) -> str:
-        raise NotImplementedError()
-
-
-class CatalogLabel(CatalogEntry):
-    def __init__(self, label: dict[str, Any]) -> None:
-        super(CatalogLabel, self).__init__()
-
-        self._l = label["attributes"]
-        self._label = label
-        self._obj_id = ObjId(self._label["id"], type=self._label["type"])
-
-    @property
-    def id(self) -> str:
-        return self._label["id"]
-
-    @property
-    def type(self) -> str:
-        return self._label["type"]
-
-    @property
-    def title(self) -> str:
-        return self._label["type"]
+        return self._e.get("description")
 
     @property
     def obj_id(self) -> ObjId:
         return self._obj_id
 
-    @property
-    def description(self) -> str:
-        return self._l["description"]
+    def __str__(self) -> str:
+        return self.__repr__()
 
+    def __repr__(self) -> str:
+        return f"CatalogFact(id={self.id}, title={self.title})"
```

Comment:
Do we really want interface to have default __repr__ for fact?

Reply:
fixed, final class name is propagated here.